api/src/main/scala/hmda/api/http/admin/InstitutionAdminHttpApi.scala

```diff
@@ -0,0 +1,82 @@
+package hmda.api.http.admin
+
+import akka.actor.{ ActorRef, ActorSystem }
+import akka.pattern.ask
+import akka.event.LoggingAdapter
+import akka.stream.ActorMaterializer
+import akka.http.scaladsl.server.Directives._
+import hmda.api.http.HmdaCustomDirectives
+import hmda.api.protocol.processing.{ ApiErrorProtocol, InstitutionProtocol }
+import hmda.model.institution.Institution
+import hmda.persistence.HmdaSupervisor.FindActorByName
+import hmda.persistence.institutions.InstitutionPersistence
+import hmda.persistence.institutions.InstitutionPersistence.{ CreateInstitution, GetInstitutionById, ModifyInstitution }
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
+import akka.http.scaladsl.marshalling.ToResponseMarshallable
+import akka.http.scaladsl.model.StatusCodes
+import akka.util.Timeout
+import hmda.api.protocol.admin.WriteInstitutionProtocol
+
+import scala.concurrent.ExecutionContext
+import scala.util.{ Failure, Success }
+
+trait InstitutionAdminHttpApi
+    extends WriteInstitutionProtocol
+    with ApiErrorProtocol
+    with HmdaCustomDirectives {
+
+  implicit val system: ActorSystem
+  implicit val materializer: ActorMaterializer
+  implicit val timeout: Timeout
+
+  val log: LoggingAdapter
+
+  val institutionsWritePath =
+    path("institutions") {
+      timedPost { uri =>
+        entity(as[Institution]) { institution =>
```

Comment:
It looks like these lines are duplicated and could be moved up outside of the `timedPost`/`timedPut`:
```scala
entity(as[Institution]) { institution =>
  extractExecutionContext { executor =>
    implicit val ec: ExecutionContext = executor
    val supervisor = system.actorSelection("/user/supervisor")
    val fInstitutionsActor = (supervisor ? FindActorByName(InstitutionPersistence.name)).mapTo[ActorRef]
```
(Though then the system wouldn't check for the appropriate verbs before it starts processing the request; not sure whether that's a concern)